### ISSUE TITLE
"Safe mode" doesn't work with hit-and-run enemies

### ISSUE THREAD

**Issue by PatrikLundell** [CleverRaven/Cataclysm-DDA#68494]
### Describe the bug

Hit-and-run tactics of some enemies has been implemented such that they are "hostile" as they approach and attack, but then turns "fleeing" as they retreat for another run. Unfortunately, "safe mode" blocks you from attacking such enemies after they've done their attack and retreat for a new attempt.

### Attach save file

N/A

### Steps to reproduce

1. Spawn a flesh raptor.
2. Wait for to get close enough to be attacked with a melee weapon, but with such a timing that it has already done its attack and thus is "fleeing".
3. Find that "safe mode" stops you from attacking the definitely hostile and dangerous enemy (until you disable "safe mode").
4. You've probably forgotten to re-enable safe mode after the mess is over, resulting in unintended attacks against critters that actually aren't hostile down the line.

### Expected behavior

The targeting of enemy creatures should not be blocked by "safe mode" just because the enemy action phase happens to be one of moving away from you.

### Versions and configuration

- OS: Windows
    - OS Version: 10.0.19045.3448 (22H2)
- Game Version: 0.G-3922-g66a0c75e4e [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Portal Storms Ignore NPCs [personal_portal_storms],
    Slowdown Fungal Growth [no_fungal_growth]
]
+ hack to disable portal storm

The version played has a local code change to mitigate wreck messes and zombie stacking, but that shouldn't be relevant.

### Additional context

I've previously seen the same thing with cockroaches that now seem to engage in hit-and-run attacks.

The only case I can think of when you want to actually be warned about attacking a fleeing enemy is when it's a human that might potentially somehow be rendered non hostile in the future. Sure, you may want to not target a fleeing mi-go to conserve ammo or to engage a more pressing target, but that's more of a tactical decision which shouldn't be interfered with by rules intended to block accidental attacks against critters that actually aren't hostile.
